scenarios/bigquery_test.py

```diff
@@ -46,10 +48,35 @@ def test_configs(self):
                 self.assertIn('metric', config)
                 self.assertIn('query', config)
                 self.assertIn('jqfilter', config)
-                bigquery.validate_metric_name(config['metric'].strip())
+                metric = config['metric'].strip()
+                bigquery.validate_metric_name(metric)
+                config_metrics.add(metric)
         configs = bigquery.all_configs()
         self.assertTrue(all(p.endswith('.yaml') for p in configs), configs)
 
+        # Check that the '**' search finds the same number of yaml
+        # files as the default search.
+        self.assertEqual(len(configs), len(config_metrics))
+
+        # Check that config files correlate with metrics listed in
+        # test-infra/metrics/README.md.
+        with open('metrics/README.md') as readme_file:
+            readme = readme_file.read()
+        start_tag = '<!-- metrics-list -->'
+        end_tag = '<!-- /metrics-list -->'
+        self.assertIn(start_tag, readme)
+        self.assertIn(end_tag, readme)
+        readme = readme[readme.index(start_tag):readme.index(end_tag)]
+        readme_metrics = set(re.findall(r'(?m)^[*-] +([\w-]+) *-.*$', readme))
+
+        missing = config_metrics - readme_metrics
+        if len(missing) > 0:
+            self.fail(msg='test-infra/metrics/README.md is missing entries'
+                      ' for metrics: {}.'.format(missing))
+        extra = readme_metrics - config_metrics
+        if len(extra) > 0:
```

Comment:
Do you intentionally want to distinguish `[]` from `None`? I don't think so, in which case please favor implicit true/false evaluation, aka `if extra:`

https://google.github.io/styleguide/pyguide.html?showone=True/False_evaluations#True/False_evaluations